packages/x-charts/src/internals/plugins/featurePlugins/useChartCartesianAxis/createAxisFilterMapper.ts

```diff
@@ -57,43 +63,80 @@ export function createAxisFilterMapper({
       return null;
     }
 
-    let extremums: number[] = [];
     const scaleType = axis.scaleType;
 
     if (scaleType === 'point' || scaleType === 'band') {
-      extremums = [0, (axis.data?.length ?? 1) - 1];
-    } else {
-      extremums = getAxisExtremum(axis, direction, seriesConfig, axisIndex, formattedSeries);
+      return createDiscreteScaleGetAxisFilter(axis.data, zoom.start, zoom.end, direction);
     }
 
-    let min: number | Date;
-    let max: number | Date;
+    return createContinuousScaleGetAxisFilter(
+      scaleType,
+      getAxisExtremum(axis, direction, seriesConfig, axisIndex, formattedSeries),
+      zoom.start,
+      zoom.end,
+      direction,
+      axis.data,
+    );
+  };
+}
 
-    const continuousScaleType =
-      !scaleType || scaleType === 'band' || scaleType === 'point' ? 'linear' : scaleType;
+export function createDiscreteScaleGetAxisFilter(
+  axisData: AxisConfig['data'],
+  zoomStart: number,
+  zoomEnd: number,
+  direction: 'x' | 'y',
+): ExtremumFilter {
+  const max = (axisData?.length ?? 1) - 1;
 
-    [min, max] = getScale(continuousScaleType, extremums, [0, 100]).nice().domain();
+  const minVal = (zoomStart * max) / 100;
+  const maxVal = (zoomEnd * max) / 100;
```

Comment:
Just to clarify what kind of max it is

```suggestion
  const maxIndex = (axisData?.length ?? 1) - 1;
  
  const minVal = (zoomStart * maxIndex) / 100;
  const maxVal = (zoomEnd * maxIndex) / 100;
```